src/Data/Hashabler/Internal.hs

```diff
@@ -626,9 +629,13 @@ instance Hashable Natural where
 #         endif
         -- Else using a BigNat (which instance calls required mixConstructor):
         (NatJ# bn)  -> hash h bn
+#   else
+        -- Natural represented with non-negative Integer:
+        n -> hash h $ toInteger n
```

Comment:
The recommended interface to `Natural` is Numeric.Natural which does not expose a constructor. Without the `integer-gmp` flag I got an error because the `Natural` constructor was not in scope. So I used the `Integral` interface.